javascript/ql/src/Declarations/UnusedVariable.ql

```diff
@@ -156,6 +156,12 @@ predicate whitelisted(UnusedLocal v) {
     isEnumMember(vd) or
     // ignore ambient declarations - too noisy
     vd.isAmbient()
+  ) or
+  exists (DirectEval eval |
+    // eval nearby
+    eval.getEnclosingStmt().getParent() = v.getADeclaration().getEnclosingStmt().getParent() and
+    // ... but not on the RHS
+    not v.getAnAssignedExpr() = eval
```

Comment:
Could you briefly motivate this criterion? I'm always a bit skeptical of conditions that involve a fixed number of `getParent`s (as opposed to a transitive closure).

We do have `DirectEval.mayAffect`, which is a much coarser approximation, but easily explained. (Personally I don't terribly mind false negatives from this query.)

Reply:
Yes, I saw `mayAffect`.
I have rebased and picked a middle ground: we now whitelist if the declaration and the `eval` call are in the same function